### ISSUE TITLE
Incorrect parent_as query generated on 3.6.2 but correct on master

### ISSUE THREAD

**Issue by dvic** [elixir-ecto/ecto#3705]
### Current behavior

For the following query, this is what v3.6.2 generates:
```elixir
query =
        from(f in "foo",
          as: :foo,
          where:
            exists(
              from(b in "bar",
                as: :bar,
                where:
                  exists(
                    from(z in "baz",
                      as: :baz,
                      where: z.z_bar == parent_as(:bar).b_id and z.zf_id == parent_as(:foo).f_id,
                      select: true
                    )
                  ),
                select: true
              )
            ),
          select: true
        )
```

```sql
SELECT TRUE
FROM "foo" AS f0
WHERE (exists((SELECT TRUE
               FROM "bar" AS sb0
               WHERE (exists((SELECT TRUE
                              FROM "baz" AS ssb0
                              WHERE ((ssb0."z_bar" = sb0."b_id") AND (ssb0."zf_id" = sb0."f_id"))))))))
```

The last condition should not be `ssb0."zf_id" = sb0."f_id"` but `ssb0."zf_id" = f0."f_id"`, it seems like it confused "bar" with "foo".

### Expected behavior

The correct table alias should be fetched or an error should be thrown.

As mentioned in the title, the master version (of ecto + ecto_sql) produces the correct query, so I assume one of the PRs fixed this? In any case, is there a new release coming soon 😇?

**Comment by josevalim** [comment]
About a month, give or take. Feel free to ping me if not by then.

**Comment by dvic** [comment]
For everyone that encounters this issue: it's fixed in v3.7.0, which came sooner than expected 👍
